### ISSUE TITLE
[V8] Plugin destroys other plugins provided typescript types 

### ISSUE THREAD

**Issue by madebyfabian** [nuxt-modules/i18n#1759]
### Environment

- Operating System: `Darwin`
- Node Version:     `v16.18.1`
- Nuxt Version:     `3.0.0`
- Nitro Version:    `1.0.0`
- Package Manager:  `npm@8.19.2`
- Builder:          `vite`
- User Config:      `modules`, `i18n`
- Runtime Modules:  `@nuxtjs/i18n@8.0.0-beta.7`
- Build Modules:    `-`

### Reproduction

Reprod without i18n plugin (working): https://github.com/madebyfabian/nuxt-reprod-i18n-ts
Reprod with i18n plugin (where the bug is): https://github.com/madebyfabian/nuxt-reprod-i18n-ts/tree/with-i18n

### Describe the bug

I have created a plain nuxt project and created an example plugin with `defineNuxtPlugin` inside `plugins/someMethod.ts`. When using it with `const { $someMethod } = useNuxtApp()`, this looks like this:

![Bildschirm­foto 2022-12-22 um 11 06 54](https://user-images.githubusercontent.com/32813692/209112013-331744f9-e587-40ff-89e5-fdcb2d57b5ec.jpg)

so typing is correctly infered.

---

After installing the i18n plugin, without even ever referencing it anywhere, so just by installing it (check the reprod branch `with-i18n` diff:
https://github.com/madebyfabian/nuxt-reprod-i18n-ts/pull/1/files

the typing of my plugins gets removed and is now `any`:

![Bildschirm­foto 2022-12-22 um 11 09 43](https://user-images.githubusercontent.com/32813692/209112485-f405a136-aa2b-4341-a47d-3a8038a302d4.jpg)

I've tried removing the complete `.nuxt` folder, recreating the generated types, but without any success. Also tried to use takover mode and the `typescript.shim = false` config, but that also doesn't change anything.

**Comment by ineshbose** [comment]
Stackblitz links:
Reprod without i18n plugin (working): https://stackblitz.com/github/madebyfabian/nuxt-reprod-i18n-ts/tree/main
Reprod with i18n plugin (where the bug is): https://stackblitz.com/github/madebyfabian/nuxt-reprod-i18n-ts/tree/with-i18n

Note: both versions are working, but the issue is about TYPE INFERENCE

**Comment by ineshbose** [comment]
The relevant files for these I think would be `.nuxt/types/plugins.d.ts` and `.nuxt/types/i18n-nuxt-app.d.ts`:

```ts
type NuxtAppInjections = 
  InjectionType<typeof import("../components.plugin").default> &
  InjectionType<typeof import("../../node_modules/nuxt/dist/head/runtime/lib/vueuse-head.plugin").default> &
  InjectionType<typeof import("../../node_modules/nuxt/dist/app/plugins/router").default> &
  InjectionType<typeof import("../../node_modules/@nuxtjs/i18n/dist/runtime/plugin").default> &
  InjectionType<typeof import("../../plugins/someMethod").default>

declare module '#app' {
  interface NuxtApp extends NuxtAppInjections { }
}
```

```ts
declare module '#app' {
  interface NuxtApp {
    $i18n: ExportedGlobalComposer & NuxtI18nRoutingCustomProperties & I18nRoutingCustomProperties
  }
}
declare module 'nuxt/dist/app/nuxt' {
  interface NuxtApp {
    $i18n: ExportedGlobalComposer & NuxtI18nRoutingCustomProperties & I18nRoutingCustomProperties
  }
}
```

Would you be able to investigate and try to isolate if either of these are causing the issue please?

**Comment by madebyfabian** [comment]
@ineshbose thanks for your quick reply! I’ll asap try to manually change those types to see which of them is causing the issue.

**Comment by madebyfabian** [comment]
So after investigation, I got the types to work:
1. Delete the `.nuxt/types/i18n-nuxt-app.d.ts` file manually
2. Remove the  `InjectionType<typeof import("../../node_modules/@nuxtjs/i18n/dist/runtime/plugin").default> &` line from the `.nuxt/types/plugins.d.ts`

Doing only one or the other does not help.

I don't have a deep knowledge about nuxt's generated types, but maybe it's because the plugin is defining/overriding the main nuxt type, and therefore all other type inferances except the i18n ones get lost.

```ts
// `../../node_modules/@nuxtjs/i18n/dist/runtime/plugin`
import { getRouteBaseName, localePath, localeRoute, switchLocalePath, localeHead } from 'vue-i18n-routing';
type GetRouteBaseName = typeof getRouteBaseName;
type LocalePath = typeof localePath;
type LocaleRoute = typeof localeRoute;
type LocaleHead = typeof localeHead;
type SwitchLocalePath = typeof switchLocalePath;
declare const _default: any;
export default _default;
declare module '#app' {
    interface NuxtApp {
        $getRouteBaseName: (...args: Parameters<GetRouteBaseName>) => ReturnType<GetRouteBaseName>;
        $localePath: (...args: Parameters<LocalePath>) => ReturnType<LocalePath>;
        $localeRoute: (...args: Parameters<LocaleRoute>) => ReturnType<LocaleRoute>;
        $localeHead: (...args: Parameters<LocaleHead>) => ReturnType<LocaleHead>;
        $switchLocalePath: (...args: Parameters<SwitchLocalePath>) => ReturnType<SwitchLocalePath>;
    }
}
declare module 'nuxt/dist/app/nuxt' {
    interface NuxtApp {
        $getRouteBaseName: (...args: Parameters<GetRouteBaseName>) => ReturnType<GetRouteBaseName>;
        $localePath: (...args: Parameters<LocalePath>) => ReturnType<LocalePath>;
        $localeRoute: (...args: Parameters<LocaleRoute>) => ReturnType<LocaleRoute>;
        $localeHead: (...args: Parameters<LocaleHead>) => ReturnType<LocaleHead>;
        $switchLocalePath: (...args: Parameters<SwitchLocalePath>) => ReturnType<SwitchLocalePath>;
    }
}

```

Hope this helped, let me know what I can do to support resolving this. I think it's interesting that no one raised this issue before, since this is easily reproducable by just using the plain nuxi starter project. Would love to help resolve this since it's currently stopping me from implementing the i18n plugin into my app (though it's a beta, I know)

**Comment by kazupon** [comment]
Hi!

This issue is resolve by edge version:
You can install `"@nuxtjs/i18n": "npm:@nuxtjs/i18n-edge"` at `package.json`
Please try it!

**Comment by kazupon** [comment]
This issue fix version will be released on v8.0.0-beta.8

**Comment by madebyfabian** [comment]
@kazupon Thanks for your work, it's indeed fixed. Will use the edge version for now.

**Comment by Husky-70** [comment]
> This issue fix version will be released on v8.0.0-beta.8
I tried the latest edge version, but unfortunately it's still not working

**Comment by Husky-70** [comment]
@kazupon I cant get it to work in the latest edge version nor in beta 8. The types from other plugins will be overwritten/destroyed by i18n. Could you confirm please?
